{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/views.py

```diff
@@ -92,43 +94,38 @@ def destroy(self, request, *args, **kwargs):
 
 
 @api_view(["post"])
-@permission_classes([])
-@authentication_classes([])
-def request_reset_link(request, *args, **kwargs):
+@permission_classes([permissions.AllowAny])
+@throttle_classes([ResetPasswordRequestLimit])
+def request_reset_code(request, *args, **kwargs):
     email = request.data.get("email")
     user = User.objects.filter(email=email).first()
     if not user:
-        return Response(status=status.HTTP_204_NO_CONTENT)
-    reset_context = user.reset_password_context()
-
-    send_html_email(
-        "Password reset for {{ cookiecutter.project_name }}",
-        "registration/password_reset.html",
-        settings.DEFAULT_FROM_EMAIL,
-        [user.email],
-        context=reset_context,
-    )
+        raise ValidationError(detail={"non_field_errors": ["User not found with that email"]})
+    UserResetPasswordCodeMessages.objects.create_code(user=user, code=random_pin_generator(count=7))
 
     return Response(status=status.HTTP_204_NO_CONTENT)
 
 
 @api_view(["post"])
 @permission_classes([permissions.AllowAny])
+@throttle_classes([ResetPasswordConfirmLimit])
 def reset_password(request, *args, **kwargs):
-    user_id = kwargs.get("uid")
-    token = kwargs.get("token")
-    user = User.objects.filter(pk=user_id).first()
-    if not user or not token:
-        raise ValidationError(detail={"non-field-error": "Invalid or expired token"})
-    is_valid = default_token_generator.check_token(user, token)
-    if not is_valid:
-        raise ValidationError(detail={"non-field-error": "Invalid or expired token"})
-    logger.info(f"Resetting password for user {user_id}")
+    email = kwargs.get("email")
+    user = User.objects.filter(email=email).first()
+    if not user:
+        raise ValidationError(detail={"non_field_errors": ["User not found with that email"]})
```

Comment:
Security issue. Someone could use this to determine what users are in the DB.

I would just fail validation with a generic error. The error should be the same regardless of the reason (ex: bad code vs bad email)